### ISSUE TITLE
Query reducer return value is not returned as prop

### ISSUE THREAD

**Issue by SachaG** [apollostack/react-apollo#429]
I haven't been able to reproduce this reliably yet (it happens with the Sidebar codebase, but not with the original [Nova]() codebase), but I've been running into a really weird behavior. My query container has this general structure:

```
const CommentsPageWithData = graphql(CommentsPageQuery, {
  props({ data }) {
     console.log(data) // 1
     // ...
     return props;
  },
  options({ params }) {
    return {
      reducer: (previousResult, action, variables) => {
         // ...
         console.log(result) // 2
         return result;
      },
    };
  },
})(CommentsPage);
```

When I execute a mutation to insert a new item, the reducer runs properly and returns an object with `n+1` items (logging out the proper value at (2)).

But when I log out the `data` object at (1), I get the *old* object before the update, with only `n` items. Which means that in turn that new correct value cannot be passed as prop to my component.

What's happening in between the reducer returning a new (correct) value and the `props` function receiving the (incorrect) result? Where is the relevant code in the `react-apollo` and `apollo-client` packages?

**Comment by SachaG** [comment]
It looks like the problem is that `currentResult.data` here is an empty object: https://github.com/apollostack/react-apollo/blob/master/src/graphql.tsx#L483

**Comment by SachaG** [comment]
Wow, I guess rubber-duck debugging really works… I was able to fix the issue by logging out `this.queryObservable` and noticing a network error in the logged out object. It turns out my mutation was using some old code and returning an incomplete fragment:

![https://d3vv6lp55qjaqc.cloudfront.net/items/2d1f1i3x1X3c3p3C1v23/Screen%20Shot%202017-01-21%20at%2016.51.03.png?X-CloudApp-Visitor-Id=9292ee42fa5c7b5a4aea538793ce9914&v=3687591e](https://d3vv6lp55qjaqc.cloudfront.net/items/2d1f1i3x1X3c3p3C1v23/Screen%20Shot%202017-01-21%20at%2016.51.03.png?X-CloudApp-Visitor-Id=9292ee42fa5c7b5a4aea538793ce9914&v=3687591e)

As far as I can tell though, this error wasn't surfaced anywhere or even available in the Redux devtools (I checked both `apollo.queries` and `apollo.mutations`). 

While I'm on the topic, I think error reporting is a big pain point of the overall Apollo developer experience. It'd be great to have something like an "Error" tab in the Apollo devtools that helps surface everything that's going wrong.
